src/site/content/en/blog/javascript-this/index.md

```diff
@@ -0,0 +1,362 @@
+console.log(boundFunction().apply({foo: 'bar'}) === boundObject);
+```
+
+When calling a bound function, the value of `this` _can't_ be changed by calling the function as a member of another object:
+
+```js
+const obj = {boundFunction};
+// Logs `true` - parent object is ignored:
+console.log(obj.boundFunction() === boundObject);
+```
+
+## Otherwise, if `this` is set at call-time: {: #call-apply }
+
+```js
+function someFunction() {
+  return this;
+}
+
+const someObject = {hello: 'world'};
+
+// Logs `true`:
+console.log(someFunction.call(someObject) === someObject);
+// Logs `true`:
+console.log(someFunction.apply(someObject) === someObject);
+```
+
+The value of `this` is the object passed to `call`/`apply`.
+
+{% Aside 'warning' %}
+Don't use `call`/`apply` to set `this` to some value unrelated to the parent object; it's usually unexpected and it's why `this` gets such a bad reputation. Consider passing the value as an argument instead; it's more explicit, and works with arrow functions.
+{% endAside %}
+
+Unfortunately `this` is set to some other value by things like DOM event listeners, and using it can result in difficult-to-understand code:
+
+{% Compare 'worse' %}
+
+```js
+element.addEventListener('click', function (event) {
+  // Logs `element`, since the DOM spec sets `this` to
+  // the element the handler is attached to.
+  console.log(this);
+});
+```
+
+{% endCompare %}
+
+I avoid using `this` in cases like above, and instead:
+
+{% Compare 'better' %}
+
+```js
+element.addEventListener('click', (event) => {
+  // Ideally, grab it from a parent scope:
+  console.log(element);
+  // But if you can't do that, get it from the event object:
+  console.log(event.currentTarget);
+});
+```
+
+{% endCompare %}
+
+## Otherwise, if the function is called via a parent object (`parent.func()`): {: #object-member }
+
+```js
+const obj = {
+  someMethod() {
+    return this;
+  },
+};
+
+// Logs `true`:
+console.log(obj.someMethod() === obj);
+```
+
+In this case the function is called as a member of `obj`, so `this` will be `obj`. This happens at call-time, so the link is broken if the function is called without its parent object, or with a different parent object:
+
+```js
+const {someMethod} = obj;
+// Logs `false`:
+console.log(someMethod() === obj);
+
+const anotherObj = {someMethod};
+// Logs `false`:
+console.log(anotherObj.someMethod() === obj);
+// Logs `true`:
+console.log(anotherObj.someMethod() === anotherObj);
+```
+
+`someMethod() === obj` is false because `someMethod` _isn't_ called as a member of `obj`. You might have encountered this gotcha when trying something like this:
+
+```js
+const $ = document.querySelector;
+// TypeError: Illegal invocation
+const el = $('.some-element');
+```
+
+This breaks because the implementation of `querySelector` looks it its own `this` value and expects it to be a DOM node of sorts, and the above breaks that connection. To achieve the above correctly:
```

Comment:
```suggestion
This breaks because the implementation of `querySelector` looks at its own `this` value and expects it to be a DOM node of sorts, and the above breaks that connection. To achieve the above correctly:
```

I think this is what you wanted to type.